cpp/CMakeLists.txt

```diff
@@ -161,9 +161,17 @@ endif()
 ################################################################################
 # - libcugraph library target --------------------------------------------------
 
+# NOTE: The most expensive compilations are listed first
+#       since ninja will run them in parallel in this order,
+#       which should give us a better parallel schedule.
+
 set(CUGRAPH_SOURCES
-    src/detail/utility_wrappers.cu
     src/detail/shuffle_wrappers.cu
+    src/sampling/random_walks_mg.cu
+    src/community/detail/common_methods_mg.cu
+    src/community/detail/common_methods_sg.cu
+    src/detail/utility_wrappers.cu
+    src/structure/graph_view_mg.cu
```

Comment:
Mainly for curiosity, why did you move graph_view_mg.cu to here?

Reply:
Ninja tracing showed that it take 217 seconds to compile.  I basically moved everything that was slow (by the unscientific method of my eyes looking at the graphic) up to the top to get it started earlier in the parallel schedule.